azpainter: 2.1.6 -> 3.0.4

dtzWill opened the pull request:
###### Motivation for this change

* github -> gitlab
* 3.0 "remade as a whole" (translated)

Release notes (back to v2.1.7): https://gitlab.com/azelpg/azpainter/-/releases?#v3.0.4
(translated:
https://gitlab-com.translate.goog/azelpg/azpainter/-/releases?_x_tr_sl=auto&_x_tr_tl=en&_x_tr_hl=en-US#v3.0.4
)

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

dtzWill committed:
azpainter: 2.1.6 -> 3.0.4

* github -> gitlab
* 3.0 "remade as a whole" (translated)
* new deps

c0bw3b reviewed (commented):
Result of `nixpkgs-review pr 157548` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>azpainter</li>
  </ul>
</details>

Builds and runs.

---

@dtzWill ~~could you change `meta.platforms` to Linux only?
It fails on Darwin apparently.~~ It probably just needs `libiconv` on Darwin

c0bw3b committed:
azpainter: add libiconv input on Darwin

and add build-time commands
and enable parallel building
and update meta.homepage

c0bw3b reviewed (approved):
Result of `nixpkgs-review pr 157548` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>azpainter</li>
  </ul>
</details>

Builds and runs
